### ISSUE TITLE
[Accessibility]: Filter connection filter label incorrect

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35192]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518, #33513

### Problem description

[This aria-label](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4d482c7b27da4ce807f3440b0e4e35441795c07e/-/blob/client/web/src/components/FilteredConnection/FilterControl.tsx?L94) is applied too specifically to all filtered connections.

### Expected behavior

Not everything is a notebook.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.
